x-pack/plugins/lens/public/pie_visualization/render_helpers.ts

```diff
@@ -44,6 +45,11 @@ export const isPartitionShape = (shape: PieChartTypes | string) =>
 export const isTreemapOrMosaicShape = (shape: PieChartTypes | string) =>
   ['treemap', 'mosaic'].includes(shape);
 
+export const shouldShowValuesInLegend = (layer: PieLayerState, shape: PieChartTypes) =>
+  layer.showValuesInLegend !== undefined
```

Comment:
Since this isn't checking the current sub type (pie, treemap, ...), it's possible to get into a weird state by enabling values in waffle, then switching to pie (without being able to disable it again):
<img width="904" alt="Screenshot 2021-12-06 at 19 08 52" src="https://user-images.githubusercontent.com/1508364/144899082-8bd8890a-be94-4cba-aeaf-865ac3db8bb5.png">

Reply:
ha.... good point, I've missed that. Thank you so much